dd-sdk-android/src/test/kotlin/com/datadog/android/rum/internal/instrumentation/gestures/GesturesListenerScrollSwipeTest.kt

```diff
@@ -435,6 +435,80 @@ internal class GesturesListenerScrollSwipeTest : AbstractGesturesListenerTest()
         verifyNoMoreInteractions(mockDatadogRumMonitor)
     }
 
+    @Test
+    fun `M use the class simple name as target name W scrollDetected { cannonicalName is null }`(
+        forge: Forge
+    ) {
+        val listSize = forge.anInt(1, 20)
+        val startDownEvent: MotionEvent = forge.getForgery()
+        val intermediaryEvents =
+            forge.aList(size = listSize) { forge.getForgery(MotionEvent::class.java) }
+        val endUpEvent = intermediaryEvents[intermediaryEvents.size - 1]
+        val expectedDirection = forge.anElementFrom(
+            GesturesListener.SCROLL_DIRECTION_DOWN,
+            GesturesListener.SCROLL_DIRECTION_UP,
+            GesturesListener.SCROLL_DIRECTION_LEFT,
+            GesturesListener.SCROLL_DIRECTION_RIGHT
+        )
+        stubStopMotionEvent(endUpEvent, startDownEvent, expectedDirection)
+        val distancesX = forge.aList(listSize) { forge.aFloat() }
+        val distancesY = forge.aList(listSize) { forge.aFloat() }
+        val velocityX = forge.aFloat()
+        val velocityY = forge.aFloat()
+        val targetId = forge.anInt()
+
+        // we will use a LocalViewClass to reproduce the behaviour when getCannonicalName function
+        // can return a null object.
+        class LocalScrollableView() : ScrollableListView()
```

Comment:
so canonical name is missing in this case because it is a local class?

Reply:
Yes....in case of a local or anonymous class this function returns `Null`